tests/test_ummg.py

```diff
@@ -0,0 +1,254 @@
+            "URL": "https://cdn.earthdata.nasa.gov/umm/granule/v1.6.4",
+            "Version": "1.6.4"
+        },
+        "Platforms": [{"ShortName": "tester-mission-9000"}],
+        "ProviderDates": [
+            {
+                "Date": "2022-08-25T21:45:44.123456Z",
+                "Type": "Insert"
+            },
+            {
+                "Date": "2022-08-25T21:45:44.123456Z",
+                "Type": "Update"
+            }
+        ],
+        "RelatedUrls": [
+            {
+                "Subtype": "VERTEX",
+                "Type": "GET DATA",
+                "URL": "http://test/test.xml"
+            },
+            {
+                "Type": "GET DATA VIA DIRECT ACCESS",
+                "URL": "s3://test/test.xml"
+            }
+        ],
+        "TemporalExtent": {
+            "RangeDateTime": {
+                "BeginningDateTime": "2022-08-25T21:45:44.123456Z",
+                "EndingDateTime": "2022-08-25T21:45:44.123456Z"
+            }
+        },
+    }
+
+
+@freeze_time("2022-08-25 21:45:44.123456")
+def test_custom_ummg():
+    class AdditionalAttributeUmmg(UmmgBase):
+        def get_additional_attributes(self) -> List[dict]:
+            return self.product_metadata["additional_attributes"]
+
+        def get_pge_version(self) -> Optional[PgeVersion]:
+            return {"PGEVersion": self.product_metadata["pge_version"]}
+
+    data = {
+        "collection_info": {
+            "short_name": "test-json-short-name",
+            "long_name": "test-json-long-name"
+        },
+        "product_md": {
+            "product_creation_datetime": datetime.now(timezone.utc),
+            "product_start_time": datetime.now(timezone.utc),
+            "product_stop_time": datetime.now(timezone.utc),
+            "mission": "tester-mission-9000",
+            "additional_attributes": [
+                {
+                    "Name": "test",
+                    "Value": "1"
+                 },
+            ],
+            "pge_version": "test-version-1"
+        },
+        "product_files_md": {
+            "product_file":  {
+                "name": "test.json",
+                "uri": "http://test/test.json",
+                "s3uri": "s3://test/test.json",
+                "size": 123456,
+                "key": "test.json",
+                "bucket": "test",
+            },
+            "related_files": [
+                {
+                    "name": "test.json.md5",
+                    "uri": "http://test/test.json.md5",
+                    "s3uri": "s3://test/test.json.md5",
+                    "size": 123,
+                    "key": "test.json.md5",
+                    "bucket": "test",
+                }
+            ]
+        }
+    }
+    ummg_data = AdditionalAttributeUmmg(data)
+    res = ummg_data.get_ummg()
+    assert res == {
+        "AdditionalAttributes": [
+            {
+                "Name": "test",
+                "Value": "1"
+            }
+        ],
```

Comment:
Doesn't match the type annotation

Reply:
I reworked the types and included a def for ummg now.